src/backend/api/meals.js

```diff
@@ -1,15 +1,121 @@
 const express = require("express");
+const { on } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    if ("maxPrice" in request.query) {
+      const maxPrice = parseFloat(request.query.maxPrice)
+      if (isNaN(maxPrice)) {
+        response.status(400).json({ error: "maxPrice must be an integer" })
+        return
+      }
+      const MaxPriceMeals = await knex('meals')
+        .where("price", "<=", maxPrice)
+      response.json(MaxPriceMeals);
+      return
+    }
+    else if ("availableReservations" in request.query) {
+      const availableReservations = request.query.availableReservations;
+      if (availableReservations == "true") {
+
+        const mealswithAvailableReservation = await knex("meals")
+          .select("meals.id", "meals.title", "meals.max_reservations ")
+          .sum({ total_reserved: 'reservations.number_of_guests' })
+          .leftJoin("reservations", "meals.id", "reservations.meal_id")
+          .groupBy("meals.title")
+          .having("meals.max_reservations", ">", "total_reserved")
+        response.json(mealswithAvailableReservation)
+        return
+      }
+    }
+    else if ("title" in request.query) {
+      const title = request.query.title.toLowerCase();
+      const matchTitle = await knex("meals")
+        .where("meals.title", "like", "%" + title + "%");
+      response.json(matchTitle);
+    }
+
+    else if ("createdAfter" in request.query) {
+      const createdAfter = new Date(request.query.createdAfter);
+      if (!createdAfter.getDate()) {
+        response.status(400).json({ error: "createdAfter should a valid date" })
+      }
+      const concertDate = await knex("meals").
+        where("created_date", ">=", createdAfter)
+      response.json(concertDate)
+    }
+
+    else if ("limit" in request.query) {
+      const limit = request.query.limit
+      console.log(limit);
+      const limitedMeals = await knex("meals").limit(limit)
+      response.json(limitedMeals);
+    }
+    else {
+      const titles = await knex("meals").select("title");
+      response.json(titles);
+    }
+  }
+
+  catch (error) {
+    console.log(error);
+  }
+});
+
+router.get("/:id", async (request, response) => {
+  try {
+    const mealID = parseInt(request.params.id);
+    if (isNaN(mealID)) {
+      response.status(400).json({ error: "MealId must be an integer" })
+      return;
+    }
+    const mealWithId = await knex('meals')
+      .where("meals.id", "=", mealID)
+    response.json(mealWithId);
+
   } catch (error) {
-    throw error;
+    console.log(error)
   }
 });
 
+router.post("/", async (request, response) => {
+
+  const insertedMeal = await knex("meals")
```

Comment:
would recommend to use a `try catch` block where you capture the error in case the insert fails and return an response to the client as an error code